readiness-tool/popup.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * @fileoverview Entry point for ART Chrome extension. 
+ * Scans the current page against the vendors in the app.json file and
+ * specifies them as either supported or not supported.
+ */
+self.popups = {};
+self.popups.isSupported = isSupported;
+self.popups.addToDict = addToDict;
+  /* @const {!Element} */
+let supportedAds;
+/* @const {!Element} */
+let supportedAnalytics;
+/* @const {!Element} */
+let notSupportedAds;
+/* @const {!Element} */
+let notSupportedAnalytics;
+/* @const {!Object} */
+let listAllApps;
+/* @const {string} */
+const loadingMessage = 'Loading...';
+/* @const {string} */
+const blankMessage = '';
+/**
+ * Callback function that sends a response upon receiving message
+ * @param {!Object} request - Message Object
+ * @param {!Object} sender - Message sender defined here
+ */
+chrome.runtime.onMessage.addListener(function (request, sender) {
+  if (request.action == 'getSource') {
+    // htmlOfTab contains the page source as a string
+    htmlOfTab = request.source;
+    findDetectedApps(htmlOfTab);
+  }
+});
+window.onload = function onWindowLoad() {
+  // When page is loaded, display 'Loading...' so the user expects content
+  supportedAds = document.getElementById('ads-supported');
+  supportedAnalytics = document.getElementById('analytics-supported');
+  notSupportedAds = document.getElementById('ads-notSupported');
+  notSupportedAnalytics = document.getElementById('analytics-notSupported');
+  supportedAds.innerHTML = supportedAnalytics.innerHTML = notSupportedAds.innerHTML = notSupportedAnalytics.innerHTML = loadingMessage;
+  // Gets the DOM of the current web page and converts it to a string
+  chrome.tabs.executeScript(null, {
+    file: 'getPagesSource.js'
+  , }, function () {});
+};
+/**
+ * Returns all the 3rd party applications found on the website
+ * @param {String} html - String containing all HTML on the page
+ * @return {Object} 
+ */
+function findDetectedApps(html) {
+  fetch('apps.json').then(function (response) {
+    if (response.status !== 200) {
```

Comment:
What if "HTTP 304 Not Modified" is returned? Generally, only 4xx and 5xx codes are real errors.

https://en.wikipedia.org/wiki/List_of_HTTP_status_codes